app/services/assessmentscores/AssessmentScoresService.scala

```diff
@@ -0,0 +1,163 @@
+      groupExercise = scores.groupExercise.map(_.copy(submittedDate = Some(dateTimeFactory.nowLocalTimeZone))),
+      leadershipExercise = scores.leadershipExercise.map(_.copy(submittedDate = Some(dateTimeFactory.nowLocalTimeZone)))
+    )
+    for {
+      _ <- assessmentScoresRepository.save(scoresWithSubmittedDate)
+      _ <- updateStatusIfNeeded(scores)
+    } yield {
+      ()
+    }
+  }
+
+  def saveExercise(applicationId: UniqueIdentifier,
+                   assessmentExerciseType: AssessmentExerciseType.AssessmentExerciseType,
+                   newExerciseScores: AssessmentScoresExercise): Future[Unit] = {
+    def updateAllExercisesWithExercise(oldAllExercisesScores: AssessmentScoresAllExercises,
+                                       newExerciseScoresWithSubmittedDate: AssessmentScoresExercise) = {
+      assessmentExerciseType match {
+        case AssessmentExerciseType.analysisExercise =>
+          oldAllExercisesScores.copy(analysisExercise = Some(newExerciseScoresWithSubmittedDate))
+        case AssessmentExerciseType.groupExercise =>
+          oldAllExercisesScores.copy(groupExercise = Some(newExerciseScoresWithSubmittedDate))
+        case AssessmentExerciseType.leadershipExercise =>
+          oldAllExercisesScores.copy(leadershipExercise = Some(newExerciseScoresWithSubmittedDate))
+      }
+    }
+
+    (for {
+      oldAllExercisesScoresMaybe <- assessmentScoresRepository.find(applicationId)
+      oldAllExercisesScores = oldAllExercisesScoresMaybe.getOrElse(AssessmentScoresAllExercises(applicationId, None, None, None))
+      newExerciseScoresWithSubmittedDate = newExerciseScores.copy(submittedDate = Some(dateTimeFactory.nowLocalTimeZone))
+      newAllExercisesScores = updateAllExercisesWithExercise(oldAllExercisesScores, newExerciseScoresWithSubmittedDate)
+      _ <- assessmentScoresRepository.save(newAllExercisesScores)
+      _ <- updateStatusIfNeeded(newAllExercisesScores)
+    } yield {
+      ()
+    })
+  }
+
+  private def updateStatusIfNeeded(allExercisesScores: AssessmentScoresAllExercises): Future[Unit] = {
+    def shouldUpdateStatus(allExercisesScores: AssessmentScoresAllExercises): Boolean = {
+      allExercisesScores.analysisExercise.map(_.isSubmitted).getOrElse(false) &&
+        allExercisesScores.groupExercise.map(_.isSubmitted).getOrElse(false) &&
+        allExercisesScores.leadershipExercise.map(_.isSubmitted).getOrElse(false)
+    }
+
+    if (shouldUpdateStatus(allExercisesScores)) {
+      applicationRepository.addProgressStatusAndUpdateAppStatus(
+        allExercisesScores.applicationId.toString(),
+        ProgressStatuses.ASSESSMENT_CENTRE_SCORES_ENTERED)
+    } else {
+      Future.successful(())
+    }
+  }
+
+  def findAssessmentScoresWithCandidateSummaryByApplicationId(applicationId: UniqueIdentifier): Future[AssessmentScoresFindResponse] = {
+    // TODO: return Option. if assessmentScores are not found, return None
```

Comment:
We will need more time to handle that and I think we want this in as soon as possible.